### PR TITLE
feat: add Dockerfile and readme instructions for Docker

### PR DESCRIPTION
**Pull request by erulabs**
Hello!

  Athens looks great, thanks for all your hard work!

  I typically run things inside a Kubernetes cluster, and as-such, I noticed the lack of a Docker image for athens-backend. I've added one in this PR and also pushed a built copy to `erulabs/athens-backend` (https://hub.docker.com/r/erulabs/athens-backend).

  I've also created a Kubernetes YAML template for Athens-backend, which is available here https://kubesail.com/template/erulabs/athens-backend

  I've seen a few notes like:

>  This is only a test deployment and all changes will be lost once heroku dyno reloads(which happens once a day atleast)

  What's the correct way to persist changes? By reading the code, it appears that `athens-sync` uses a path like "~/athens/db/athens-sync". If that's correct, I'll setup the Kubernetes example and docker example to include that as a "Volume", so changes won't be lost on restarts of the service. **EDIT**: I've gone ahead and added a `VOLUME` for `/root/athens/db/athens-sync` in the Dockerfile as well as a `PersistentVolumeClaim` in the [YAML](https://kubesail.com/template/erulabs/athens-backend).

`<pitch>`A small note: I'm the founder of KubeSail.com - we essentially make it easy for "self-hosted" products like Athens to easily build a hosted version for their customers - think Shopify but for Docker containers - and to easily handle on-prem and enterprise installations. I see ya'll are working on a hosted version of Athens - we can get this going in a matter of hours, instead of months. Feel free to reach out!`</pitch>`

Hopefully this helps people run athens-backend a bit easier! No Clojure/Lein install required 💃

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM clojure:openjdk-11-lein
+
+ENV HTTP_PORT "13337"
+
+ADD . .
+
+RUN mkdir -p ~/athens/db/athens-sync && \
+  lein install
```
Comment (neotyk) [inline Dockerfile:8]: `lein uberjar` will work better.
I think `lein install` just packages jar, but doesn't AOT code.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+EXPOSE 13337
+
+VOLUME ["/root/athens/db/athens-sync"]
+
+ENTRYPOINT ["lein", "run"]
```
Comment (neotyk) [inline Dockerfile:14]: Since we should have uberjar available at this point, let's use it.

File: README.md [hunk 3]
```diff
@@ -24,3 +26,11 @@ Steps to test out **Real time collaboration**
...
 
 This is only a **test deployment** and **all changes will be lost** once heroku dyno reloads(which happens once a day atleast)
+
+### Docker
+
+1. Clone the repo
+2. `docker build -t athens-backend .`
+3. `docker run -it -p 13337:13337 athens-backend`
+
+Note that the Docker container exposes "13337" by default instead of "1337" so that this container can run in less privileged environments which may not allow low-ports. It can still be overridden by the HTTP_PORT variable.
```
Comment (neotyk) [inline README.md:36]: I get the port change, it would be nice to also change default value in code as well.

File: README.md [hunk 4]
```diff
@@ -24,3 +26,11 @@ Steps to test out **Real time collaboration**
 #### Important Note
 
 This is only a **test deployment** and **all changes will be lost** once heroku dyno reloads(which happens once a day atleast)
+
+### Docker
+
+1. Clone the repo
+2. `docker build -t athens-backend .`
```
Comment (neotyk) [inline README.md:33]: Could we also get docker-compose for it?
Now we're using only FS, but before we release I'd like to see if we could persist in PostgreSQL.

**Review by neotyk — COMMENTED** [review]
Thank you for your contribution @erulabs 
Left few comments.

I've started working on Makefile for this build here: https://github.com/athensresearch/athens-backend/tree/dev/build
Exactly with idea to use it for docker build, would you like a makefile for it?

File: Dockerfile (hunk 1 above)
Comment (erulabs) [reply Dockerfile]: Ah, good to know - I'm not familiar with Clojure so this is helpful - replaced with `uberjar`

File: Dockerfile (hunk 2 above)
Comment (erulabs) [reply Dockerfile:14]: Is there something preferred to `lein run`? This seems to work with `uberjar`, although not sure if there is a `run-jar` or something similar

File: README.md (hunk 4 above)
Comment (erulabs) [reply README.md:33]: Added docker-compose.yml

File: Dockerfile (hunk 2 above)
Comment (tangjeff0) [reply Dockerfile:14]: @erulabs running `lein uberjar` compiles two `jar` files:

`athens-sync/target/uberjar/athens-sync-0.1.0-SNAPSHOT.jar`
`athens-sync/target/uberjar/athens-backend.jar`

You can then move these jar files to a VM and run them assuming there is a Java run-time (need to install Java, but not Lein and Clojure).

File: README.md (hunk 3 above)
Comment (tangjeff0) [reply README.md:36]: I think I get this too. Containers generally have high ports. What about when using port `80` on a normal EC2 instance?

File: Dockerfile (hunk 2 above)
Comment (tangjeff0) [reply Dockerfile:14]: Therefore the final command should probably be something like 

`java -jar ./target/uberjar/athens-backend.jar` which translates to

`ENTRYPOINT ["java", "-jar", "./target/uberjar/athens-backend.jar"]`

Not sure how specifying port would work in this case. Maybe some args/env that need to be set. Interface is  `java -jar ./target/uberjar/athens-backend.jar -p 12345` in any case

Comment (tangjeff0) [reply Dockerfile:14]: My docker build outputs

```
Created /tmp/target/uberjar/athens-sync-0.1.0-SNAPSHOT.jar
Created /tmp/target/uberjar/athens-backend.jar
```

but the ENTRYPOINT ends up running, so just gonna merge this and say it works!

**Review by tangjeff0 — APPROVED** [review]

**Pull request merged**